src/modules/proposal/types/history-event-data-map.type.ts

```diff
@@ -0,0 +1,53 @@
+import { HistoryEventType } from '../enums/history-event.enum';
+
+// Add imports for any types needed for data shapes here
+
+export type HistoryEventDataMap = {
```

Comment:
This might be line saving:

```ts
type HistoryEventData = {
  [HistoryEventType.ParticipantAdded]: { participantName: string };
  [HistoryEventType.ParticipantRemoved]: { participantName: string };

  // Add more mappings here as needed
};

export type HistoryEventDataMap<T extends HistoryEventType> = T extends keyof HistoryEventData
  ? HistoryEventData[T]
  : undefined;
```

Requires another change:
```ts
// src/modules/proposal/utils/proposal-history.util.ts

const pushHistoryItem = <T extends HistoryEventType>(
  proposalAfterChanges: Proposal,
  user: IRequestUser,
  type: T,
  location?: MiiLocation,
  data?: HistoryEventDataMap<T>, // <----Here
```